src/components/Discussion/Internal/Comment/Body.js

```diff
@@ -0,0 +1,173 @@
+import { renderMdast } from 'mdast-react-render'
+
+import { DiscussionContext } from '../../DiscussionContext'
+
+import { Label } from '../../../Typography'
+import { sansSerifRegular14 } from '../../../Typography/styles'
+
+import colors from '../../../../theme/colors'
+import { mUp } from '../../../../theme/mediaQueries'
+import { useMediaQuery } from '../../../../lib/useMediaQuery'
+import { useSize } from '../../../../lib/useSize'
+
+import createCommentSchema from '../../../../templates/Comment'
+
+import { Context } from './Context'
+
+import { COLLAPSED_HEIGHT } from '../../config'
+
+const schema = createCommentSchema()
+const highlightPadding = 7
+
+const buttonStyle = {
+  outline: 'none',
+  WebkitAppearance: 'none',
+  background: 'transparent',
+  border: 'none',
+  padding: '0',
+  display: 'block',
+  cursor: 'pointer',
+  height: '100%'
+}
+
+const styles = {
+  container: css({
+    position: 'relative'
+  }),
+  highlight: css({
+    top: -highlightPadding,
+    left: -highlightPadding,
+    right: -highlightPadding,
+    bottom: -highlightPadding,
+    padding: highlightPadding,
+    width: `calc(100% + ${highlightPadding * 2}px)`,
+    backgroundColor: colors.primaryBg
+  }),
+  margin: css({
+    display: 'block',
+    marginTop: 8
+  }),
+  unpublished: css({
+    marginBottom: 8
+  }),
+  collapsedBody: css({
+    height: `${COLLAPSED_HEIGHT.mobile}px`,
+    overflow: 'hidden',
+    [mUp]: {
+      maxHeight: `${COLLAPSED_HEIGHT.desktop}px`
+    }
+  }),
+  context: css({
+    marginBottom: 10
+  }),
+  collapeToggleContainer: css({
+    borderTop: `1px solid ${colors.divider}`,
+    marginTop: 6
+  }),
+  collapseToggleButton: css({
+    ...buttonStyle,
+    ...sansSerifRegular14,
+    color: colors.primary,
+    height: '32px',
+    lineHeight: '32px',
+    '&:hover': {
+      color: colors.secondary
+    }
+  })
+}
+
+const MissingNode = ({ node, children }) => {
+  return (
+    <span
+      style={{
+        textDecoration: `underline wavy ${colors.divider}`,
+        display: 'inline-block',
+        margin: 4
+      }}
+      title={`Markdown element "${node.type}" wird nicht unterstützt.`}
+    >
+      {children || node.value || node.identifier || '[…]'}
+    </span>
+  )
+}
+
+export const Body = ({ t, comment, context }) => {
+  const { discussion, highlightedCommentId } = React.useContext(DiscussionContext)
+  const { collapsable } = discussion
+
+  const { published, content, userCanEdit, adminUnpublished } = comment
+
+  /**
+   * Measuring the comment body size (height), so we can deterimine whether to collapse
+   * the comment body.
+   *
+   * bodyVisibility:
+   *   - 'indeterminate': We don't know yet whether to collapse the body or not.
+   *   - 'full': The body is collapsable but we're showing the full body.
+   *   - 'preview': The body is collapsed.
+   */
+  const [bodyVisibility, setBodyVisibility] = React.useState('indeterminate')
+  const [bodyRef, bodySize] = useSize()
```

Comment:
Use an `useEffect(() => {}, [content])` that simply measures the body once after rendering different content instead. I don't see any need for a resize observer.

This of course can be abstracted as `useMeasureOnce` or something like that. Or even call it `useSize`, if you want, but with content change dependency ;)

Reply:
I created a new hook: `useBoundingClientRect()`. The old hook is still available, in case you want to use it at some point in the future.